lib/offline.js

```diff
@@ -119,19 +131,30 @@ module.exports = function(config) {
         var data = fs.readFileSync(filepath);
         totalGzipSize += gzipSize.sync(data);
         var hash = getHash(data);
-        process.stdout.write(chalk.green.bold('✓ ') + 'Caching ' + filepath + ' (' + prettyBytes(size) + ')\n');
+        var path = getPathName(filepath, rootDir);
+        var rootMatch = new RegExp('^' + rootDir);
+        var pathAtString = '';
+        if (filepath.replace(rootMatch, './') !== path) {
+          pathAtString = ' at path ' + path;
+        }
+
+        process.stdout.write(chalk.green.bold('✓ ') + 'Caching ' + filepath + pathAtString + ' (' + prettyBytes(size) + ')\n');
         return {
-          path: filepath.replace(rootDir, function (match, offset) {
-            // The root must be the worker's directory
-            return offset === 0 ? './' : match;
-          }),
+          path: path,
           hash: hash,
         };
       });
 
       process.stdout.write('Total cache size is ' + prettyBytes(totalSize) + ' (' + prettyBytes(totalGzipSize) + ' if served with gzip) for ' + files.length + ' files.\n');
 
-      return filesAndHashes;
+      filesAndHashes.unshift({
+        path: './'
+      }); // cache always the current root to make the default page available
+
+      // Remove duplicates from file list
+      return filesAndHashes.filter(function(fileName, index) {
+        return filesAndHashes.indexOf(fileName) === index;
+      });
```

Comment:
This doesn't seem to be working, as I still see two occurrences of `'./'` in the list of *RESOURCES* when I offline a project using this branch:

>    RESOURCES: [
>      './', // 
>      …
>      './', // 238fefe39dae24af2aa4b62f61f3745953b4858c

It also doesn't seem like the right one would win in this case, since the one without the hash appears first in the list and would presumably be selected, which drops information.